ProjectSidewalk/SidewalkWebpage
# 1214 assign region from priority

This PR updates the code that selects and assigns regions to make use of our new priority-based system.

The code is used when a new user comes to the site and needs to be assigned a neighborhood to audit, when a user finishes auditing a neighborhood completely, and any time an anonymous user visits the audit page.

Previously, a user would be assigned to one of the neighborhoods that had the lowest completion percentage. Now a user is assigned to a neighborhood with one of the 5 highest average priorities across its streets.

Testing:
I added print statements in the relevant places, so when you run each case, note the output of the print statements.
1. Visit the audit page as an anonymous user. You should see the following print statements
    EASY SELECTION
    GENERIC SELECTION
    Highest priority regions: List(r1, r2, r3, r4, r5)
2. Sign in, visit audit/nextRegion=easy. You should see the following print statements (note that for users, there is a step where we _assign_ them this region for the future, which is why you see that additional line printed; also note that the list of high priority regions should be the same, unless you have audited one of those regions completely yourself)
    EASY ASSIGNMENT FOR USER
    EASY SELECTION FOR USER
    GENERIC SELECTION
    Highest priority regions: List(r1, r2, r3, r4, r5)
3. Stay signed in, visit audit/nextRegion=regular. You should see the following print statements (there may be one or two different regions in this list, which would only happen any of the highest priority regions are "difficult" regions)
    REGULAR ASSIGNMENT FOR USER
    REGULAR SELECTION FOR USER
    GENERIC SELECTION
    Highest priority regions: List(r1, r2, r3, r4, r5)

So these are the 3 cases that we would actually expect to see while people use the tool. Most of the functions have safeguards in case no regions are found initially. For example, the function for selecting an easy region checks if something was successfully selected; if not, it calls the regular selection function instead, since the regular selection is less restrictive. The same thing goes for the functions that select based on regions that the user has not completed; if none are found, we default to the functions that assign any region.

To test some of these backups, you can go into the functions in RegionTable.scala and modify the list of possible region ids, causing the function to go to a backup. For example, you could go to line 141, where the list of region ids for easy regions that the user has not completely audited is created. You could replace it with an empty list. Then when you run the audit/nextRegion=easy route, you will see the print statements for selecting an easy region, but that fails and calls the regular selection code, so you will also see print statements for that case as well.

Resolves #1214
